### ISSUE TITLE
ERROR: Failed building wheel for gym

### ISSUE THREAD

**Issue by J-cyber0** [freqtrade/freqtrade#8376]
## Describe your environment

  * Operating system: Ubuntu 20.04.6
  * Python Version: 3.8.10
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

Got this error after executing the ./setup.sh -i command in a WSL terminal. Looks like a simple fix for the invalid version number of 'opencv-python' module. 

### Steps to reproduce:

  1. Open WSL terminal
  2. Successfully complete Freqtrade installation instructions up to the command './setup.sh -i'.
  3. Execute the './setup.sh -i' command
  
### Observed Results:

  * Successfully built freqtrade but failed to build wheel for gym
  * build wheel for gym

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
  // Traceback (most recent call last):
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/requirements.py", line 35, in __init__
          parsed = parse_requirement(requirement_string)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/_parser.py", line 64, in parse_requirement
          return _parse_requirement(Tokenizer(source, rules=DEFAULT_RULES))
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/_parser.py", line 82, in _parse_requirement
          url, specifier, marker = _parse_requirement_details(tokenizer)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/_parser.py", line 126, in _parse_requirement_details
          marker = _parse_requirement_marker(
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/_parser.py", line 147, in _parse_requirement_marker
          tokenizer.raise_syntax_error(
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/_tokenizer.py", line 163, in raise_syntax_error
          raise ParserSyntaxError(
      wheel.vendored.packaging._tokenizer.ParserSyntaxError: Expected end or semicolon (after version specifier)
          opencv-python>=3.
                       ~~~^

      The above exception was the direct cause of the following exception:

      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-90bmqlq8/gym_617fad43752440b291159ab25049b57c/setup.py", line 39, in <module>
          setup(
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
          self.run_command(cmd)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 395, in run
          self.egg2dist(self.egginfo_dir, distinfo_dir)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 534, in egg2dist
          pkg_info = pkginfo_to_metadata(egginfo_path, pkginfo_path)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/metadata.py", line 160, in pkginfo_to_metadata
          for key, value in generate_requirements({extra: reqs}):
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/metadata.py", line 138, in generate_requirements
          for new_req in convert_requirements(depends):
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/metadata.py", line 103, in convert_requirements
          parsed_requirement = Requirement(req)
        File "/home/practice/freqtrade/.env/lib/python3.10/site-packages/wheel/vendored/packaging/requirements.py", line 37, in __init__
          raise InvalidRequirement(str(e)) from e
      wheel.vendored.packaging.requirements.InvalidRequirement: Expected end or semicolon (after version specifier)
          opencv-python>=3.
                       ~~~^
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip
  ```

**Comment by xmatthias** [comment]
If you think it's a problem in gym - then that's best reported there.

The truncated / reduced log does not include enough information to determine anything (the exception doesn't show it's caused by gym).

Now that said - `opencv-python` is not a package freqtrade depends on (neither does gym unless it's installed with extras, which we don't)  - and is not a package i even have installed (on a full freqtrade installation including all dependencies).

**Comment by xmatthias** [comment]
One sidenote - considering that your system is rather old (ubuntu 22.04 would be an uptodate WSL LTS version), you might want to consider running a system update first (`sudo apt update && sudo apt upgrade`).

Please also check for updates to pip and packaging (`pip3 install -U pip packaging`).

**Comment by FreakyHead** [comment]
Same for me

Python:                 3.9.16
Pip:                        23.0.1
setuptools:           65.5.0

------------------------

ProductName:      macOS
ProductVersion:   13.2.1
BuildVersion:	      22D68


Log in attachments

[log.txt](https://github.com/freqtrade/freqtrade/files/11123234/logs.txt)

**Comment by FreakyHead** [comment]
I solved the problem by these steps:

#### 1. Install the required build tools if they are not already installed:
``` 
brew install cmake
brew install libomp
```

#### 2. Install lightgbm manually using pip:
```
pip install lightgbm
```

#### 3. Repeat installation:
```
./setup.sh -i
```


### Successfully built freqtrade, but it still has errors in the console ☹️ : 
[logs1.txt](https://github.com/freqtrade/freqtrade/files/11123377/logs1.txt)

**Comment by J-cyber0** [comment]
> I solved the problem by these steps:
> 
> #### 1. Install the required build tools if they are not already installed:
> ```
> brew install cmake
> brew install libomp
> ```
> 
> #### 2. Install lightgbm manually using pip:
> ```
> pip install lightgbm
> ```
> 
> #### 3. Repeat installation:
> ```
> ./setup.sh -i
> ```
> 
> ### Successfully built freqtrade, but it still has errors in the console ☹️ :
> [logs1.txt](https://github.com/freqtrade/freqtrade/files/11123377/logs1.txt)

Thank  you, but it looks like you still received the initial error, as well as I am not suffering or noticing any deficiency with FreqAI as of yet and I do believe this is a gym issue as xmatthias has mentioned. I've yet to dabble with the Reinforcement Learning feature but I suspect this will be the area where the failed gym wheel could manifest, but till then I will bring this issue to gym.

**Comment by J-cyber0** [comment]
> One sidenote - considering that your system is rather old (ubuntu 22.04 would be an uptodate WSL LTS version), you might want to consider running a system update first (`sudo apt update && sudo apt upgrade`).
> 
> Please also check for updates to pip and packaging (`pip3 install -U pip packaging`).

Thank you. Will do this.

**Comment by xmatthias** [comment]
well gym won't "really" fix this.
unitl stable baselines releases their (final) update updating to gymnasium, we're stuck with an outdated version of gym.
it's not the latest version of gym - and the latest version is superceded by gymnasium, too, both of which are currently not supported by stable-baselines.

once this dependency mess clears, this should work better as the new packages all have wheels available afaik, or at least correspond to current packaging rules.

**Comment by STraderMX** [comment]
I don't know if you have solve this, but I have fixed it by forcing `wheel` to `0.38.4`, the best way is by forcing it in the `setup.sh`, like this:  
<img width="483" alt="image" src="https://user-images.githubusercontent.com/122330178/233880416-c4ebc655-1a1d-4de2-9767-1ff28c30bd27.png">

**Comment by hom-bahrani** [comment]
> I don't know if you have solve this, but I have fixed it by forcing `wheel` to `0.38.4`, the best way is by forcing it in the `setup.sh`, like this: <img alt="image" width="483" src="https://user-images.githubusercontent.com/122330178/233880416-c4ebc655-1a1d-4de2-9767-1ff28c30bd27.png">

Thanks this resolved the same installation issue for me. I was installing on Ubuntu 20  with python 3.8

**Comment by Dorus-coder** [comment]
> > I don't know if you have solve this, but I have fixed it by forcing `wheel` to `0.38.4`, the best way is by forcing it in the `setup.sh`, like this: <img alt="image" width="483" src="https://user-images.githubusercontent.com/122330178/233880416-c4ebc655-1a1d-4de2-9767-1ff28c30bd27.png">
> 
> Thanks this resolved the same installation issue for me. I was installing on Ubuntu 20 with python 3.8

This solved my issues as well. I was installing on Windows 10 with python 3.10.

However I'm encountering the following warning. Could this be a problem?
WARNING: Ignoring invalid distribution -heel (c:\users\cursist\dorus\opengym\venv\lib\site-packages)

**Comment by J-cyber0** [comment]
looks like you misspelled 'wheel'?

**Comment by spirodelazz** [comment]
> 

Thanks! I was installing on MacBook Pro M1 with python3.10.

**Comment by reza-naghizadeh** [comment]
> I don't know if you have solve this, but I have fixed it by forcing `wheel` to `0.38.4`, the best way is by forcing it in the `setup.sh`, like this: <img alt="image" width="483" src="https://user-images.githubusercontent.com/122330178/233880416-c4ebc655-1a1d-4de2-9767-1ff28c30bd27.png">

i was facing a problem for installing gym-pybullet-drone and this solved it thank you alot❤️❤️❤️

**Comment by PavanMantripragada** [comment]
> > I don't know if you have solve this, but I have fixed it by forcing `wheel` to `0.38.4`, the best way is by forcing it in the `setup.sh`, like this: <img alt="image" width="483" src="https://user-images.githubusercontent.com/122330178/233880416-c4ebc655-1a1d-4de2-9767-1ff28c30bd27.png">
> 
> i was facing a problem for installing gym-pybullet-drone and this solved it thank you alot❤️❤️❤️

This solved my issue too! upgrading wheel (not just setuptools) was imp!
